charts/k8up/templates/cleanup-hook.yaml

```
@@ -86,14 +86,16 @@ spec:
       containers:
       - name: "{{ .Release.Name }}-cleanup"
         image: "{{ include "cleanupImage" . }}"
-        imagePullPolicy: {{ .Values.cleanup.pullPolicy }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
         command:
           - sh
           - -c
         args:
           - |
             #!/bin/bash
 
+            set -e
```

This will not work as expected. The `jq` command masks the exit code of kubectl.

```suggestion
            set -e

           NAMESPACE_JSON=$(kubectl get namespace -ojson)
           NAMESPACES=$(printf "%s" "$NAMESPACE_JSON" | jq -r '.items[].metadata.name')
```

or use `-o pipefail`. As far as I can tell alpine has `ash` as the defailt shell which should support the pipefail option.

The shebang can be removed as it's not effective if used through `-c`.

---

Pipefail is fine:

```
❯ docker run --rm -it alpine:3.20
/ # sh -c 'set -e ; false|true; :' ; echo "$?"
0
/ # sh -c 'set -eo pipefail ; false|true; :' ; echo "$?"
1
/ # sh -c 'set -eo pipefail ; true|true; :' ; echo "$?"
0
```